pkg/maps/ctmap/gc/gc.go

```diff
@@ -37,7 +38,8 @@ type PerClusterCTMapsRetriever func() []*ctmap.Map
 type parameters struct {
 	cell.In
 
-	Logger logrus.FieldLogger
+	Context context.Context
```

Comment:
I don't think this is a good pattern in hive. The `cell.In` marker (see [docs](https://docs.cilium.io/en/stable/contributing/development/hive/#provide)) marks this struct as the dependencies to be injected by hive. Contrary to a logger or the k8s clientset (https://docs.cilium.io/en/stable/contributing/development/hive/#cell-showcase), we don't have a global context passed in by hive. Instead, use the Start and Stop hooks that hive provides, via https://docs.cilium.io/en/stable/contributing/development/hive/#lifecycle.

Reply:
:+1:, I've gotten ridden of it. See https://github.com/cilium/cilium/pull/28183#discussion_r1339416931.